### ISSUE TITLE
Better rendering of Pretty format with columns of a large width

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#59502]
Here is an example from GlareDB - it is a slow and basically a non-functional toy, but they made the rendering of the Pretty format better than ClickHouse did:

![Screenshot_20240202_012254](https://github.com/ClickHouse/ClickHouse/assets/18581488/6fb95edf-6ee9-4797-b9e5-7692a13f979e)

![Screenshot_20240202_012355](https://github.com/ClickHouse/ClickHouse/assets/18581488/178a520b-67d0-4d91-bd39-e6be3d7d538e)

**Comment by al13n321** [comment]
I guess this should be a separate format, not replacement for Pretty, it hides too much information. Sure the user would enjoy the eye candy on their first few queries, but it would be outweighed by annoyed googling as soon as they need to get any actual information out of the DB.

* Truncating numbers usually makes them useless. I think only unbounded-length types (strings and arrays) should be truncated.
* Truncating strings to title length when there are lots of columns looks nice. But there should be some easily-discoverable way to see the whole string too. Same for …-ing the middle rows (20 is very few, I often scroll for a few hundred). Same for …-ing the middle columns.
* Seems better to omit the last columns instead of the middle, and have a convention to put the most useful columns at the beginning when creating tables (and have an easy and efficient ALTER to reorder columns). This would make things like `select * from system.parts` readable and useful.
* +4 lines of header in order to show data types - I guess it's worth it. The `—` line seems unnecessary though.

**Comment by alexey-milovidov** [comment]
Yes, I also found the behavior of GlareDB ridiculous. Most likely, it uses an existing Rust crate from some enthusiast.

Let's make the following changes:
1. Not truncate it horizontally, only vertically.
2. Don't use this minuscule size of 25 by default, but also less than our current default 10,000 - I think, 500 will be ok.
3. If the result is large, duplicate the header also as a footer.

**Comment by alexey-milovidov** [comment]
Done.
